src/vm/docs/mod.rs

```diff
@@ -358,6 +358,34 @@ value return by the successive applications.",
 (fold * (list 2 2 2) 0) ;; Returns 0"
 };
 
+const CONCAT_API: SpecialAPI = SpecialAPI {
+    input_type: "(buff, buff)|(list, list)",
+    output_type: "buff|list",
+    signature: "(concat buff-a buff-b)",
+    description: "The `concat` function takes two buffers or two lists with the same entry type, 
+and returns a concatenated buffer or list of the same entry type, with max_len = max_len_a + max_len_b.",
+    example: "(concat \"hello \" \"world\") ;; Returns \"hello world\""
+};
+
+const APPEND_API: SpecialAPI = SpecialAPI {
+    input_type: "list A, A",
+    output_type: "list",
+    signature: "(append (list 1 2 3 4) 5)",
+    description: "The `append` function takes a list and another value with the same entry type, 
+or a buffer and another buffer of length 1 and outputs a buffer or a list of the same type with max_len += 1.",
+    example: "(append (list 1 2 3 4) 5) ;; Returns (list 1 2 3 4 5)"
+};
+
+const ASSERTS_MAX_LEN_API: SpecialAPI = SpecialAPI {
+    input_type: "buff|list, uint",
+    output_type: "(optional buff|list)",
+    signature: "(asserts-max-len! buffer 10)",
+    description: "The `asserts-max-len!` function takes a length N (must be a literal) and a buffer or list argument, which must be typed as a list 
+or buffer of length M and outputs that same list or buffer, but typed with max length N. 
+At runtime, a check is performed, which if it fails, returns a (none) option.",
+    example: "(asserts-max-len! (list 2 2 2) 3) ;; Returns (list 2 2 2)"
```

Comment:
Wouldn't this return `(some (list 2 2 2))`?

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1139/commits/387da504ed2d0af7da4b3eda5092bdf6cfb8f8ce